File: dataAccess/src/main/java/da/factor/FactorService.java
package da.factor;

import java.util.List;

public interface FactorService {
	
	public List<Factor> findAll();
	
	public Factor findOne(Long id);
	
	public Factor save(Factor factor);
	
	public void delete(Long id);

	public List<Factor> findByCategory(Long categoryId);

}

File: external/src/main/webapp/external-view/src/app/factor/factor.service.ts
import { Injectable } from '@angular/core';
import { Http, Response } from "@angular/http";
import 'rxjs/add/operator/map'
import 'rxjs/add/operator/catch';
import { Subject } from "rxjs/Subject";

import { Factor } from './factor'
import { environment } from '../../environments/environment';

@Injectable()
export class FactorService {

    private apiUrl = `${environment.BACKEND_URL}/factor`;

    constructor(private http: Http) { }

    findByCategory(categoryId) {
        return this.http.get(this.apiUrl + '/category/' + categoryId)
            .map(res => res.json());
    }
}
